core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -114,6 +116,9 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
         raf.setLength(roundedNewSize)
         mmap = raf.getChannel().map(FileChannel.MapMode.READ_WRITE, 0, roundedNewSize)
         _maxEntries = mmap.limit / entrySize
+        if(newlyCreated){
+          mmap.put(contentBuffer.asInstanceOf[ByteBuffer])
```

Comment:
On windows the original memory map pointed to by contentBuffer would have been unmapped at this point.